main.ts

```diff
@@ -153,8 +153,13 @@ export default class FitPlugin extends Plugin {
 				}
 				return true
 			}
-			console.error("Caught unknown error: ", error)
-			notice.setMessage("Unable to sync, if you are not connected to the internet, turn off auto sync.", true)
+      console.error("Caught unknown error: ", error)
+      
+      try {
+          notice.setMessage(`Unable to sync, if you are not connected to the internet, turn off auto sync. Error: ${error.message}`, true)
+      } catch (e) {
+          notice.setMessage("Unable to sync, if you are not connected to the internet, turn off auto sync.", true)
+      }
```

Comment:
This should ensure to always have a `message`

```suggestion
          console.error("Caught unknown error: ", error)
          if (!(error instanceof Error)) {
              error = new Error(error);
          }
          notice.setMessage(`Unable to sync, if you are not connected to the internet, turn off auto sync. Error: ${error.message}`, true)
```